includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -2343,4 +2464,161 @@ protected function guess_modal_role( DOMElement $modal ) {
 		// None of the roles we are looking for match any of the classes.
 		return Role::DIALOG;
 	}
+
+	/**
+	 * Evaluates the given XPath expression and give first Element if exist.
+	 *
+	 * @param string  $expression   The XPath expression to execute.
+	 * @param Element $context_node The optional context node can be specified for doing relative XPath queries.
+	 *                              By default, the queries are relative to the root element.
+	 *
+	 * @return Element|null Return Element if exists otherwise null.
+	 */
+	protected function get_first_element( $expression, $context_node = null ) {
+
+		$dom_node_list = $this->dom->xpath->query( $expression, $context_node );
+
+		/** @var Element $dom_node */
+		$dom_node = $dom_node_list->item( 0 );
+
+		return $dom_node;
```

Comment:
It could also be `null` here. So maybe to be more explicit:

```suggestion
		$dom_node = $dom_node_list->item( 0 );
		return $dom_node instanceof Element : $dom_node : null;
```

Reply:
To be even more robust, it could make sure that `$dom_node_list` is a `NodeList`.